### DISCUSSION TITLE
File Library - change view mode request, modify button functionality request and feature request

### DISCUSSION THREAD

**Discussion by kzsa** [directus/directus#7616] [Feature Requests]
Hi,
Thank you very much for all your work.

I would like to ask you to consider introducing the following options:

### Feature request
- "File Library Cloud" / "Remote Links Collection" near "File directory"
There would be many advantages to having a "Remote Links Collection" next to the "File Library" (possibility to add ftp folders / links to files / folders pointing to the central file server), so that you don't have to unnecessarily duplicating them (maintenance, etc.)

### Modify button functionality request
- Asside > (module-nav-content)
If I click on the folder names, the underneathed subfolders are not displayed. To display the subfolders, I have to use the ">" icon. During my tests, 90% of my users did not realize that there was a folder underneath. It bothered me too, even though I knew how it worked.
Clicking on ">" will add the "ul" in real time, but I don't think this is a good approach, as the folder represented by the element already exists, so it should be treated as a "permanent" / existing menu item, rather than added and removed from the tree each time. Perhaps it would be more elegant and practical to affect the element's visibility via local CSS variables. I use the scale and opacity properties in such cases.
If this would be changed, it would also benefit the subject of my next question.

### Change view mode request
- "File Lybrary" / folders of assets
It would be very convenient if not only files, but also folders could be displayed in the view, like in a classic file manager.
Maybe this feature could be optional.

Thank you for the opportunity to share some of my thoughts with you.
Best regards: kzsa

**Comment by benhaynes** [comment]
I agree that folders should be included in the file listing layout view. We have a design for that if someone wants to try a PR.

**Comment by kzsa** [comment]
Hi Ben,

I would like to try it, if there is a way! I am currently testing 9.0.0-rc.91 on a Vagrant box. I really like it! 🤩 I feel the wind of freedom again. 

Br: kzsa
